File: src/types/terminal.ts
export type TerminalStatus = 'idle' | 'running' | 'completed' | 'error';

export interface TerminalSession {
  session_id: string;
  project_id: string;
  command: string | null;
  status: TerminalStatus;
  pid: number | null;
}

export interface TerminalLogMessage {
  session_id: string;
  project_id: string;
  message: string;
  stream: 'stdout' | 'stderr';
}

File: src-tauri/src/models/terminal.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum TerminalStatus {
    Idle,
    Running,
    Completed,
    Error,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TerminalSession {
    pub session_id: String,
    pub project_id: String,
    pub command: Option<String>,
    pub status: TerminalStatus,
    pub pid: Option<u32>,
}
